ebml_matroska.xml

```diff
@@ -347,29 +347,29 @@ to define how the BlockAdditional data should be handled.</documentation>
     <documentation lang="en" purpose="definition">Extra binary data that the BlockAddIDType can use to interpret the BlockAdditional data.
 The interpretation of the binary data depends on the BlockAddIDType value and the corresponding Block Additional Mapping.</documentation>
   </element>
-  <element name="Name" path="\Segment\Tracks\TrackEntry\Name" id="0x536E" type="utf-8" maxOccurs="1">
+  <element name="Name" path="\Segment\Tracks\TrackEntry\Name" id="0x536E" type="utf-8" length="not 0" maxOccurs="1">
     <documentation lang="en" purpose="definition">A human-readable track name.</documentation>
     <extension type="libmatroska" cppname="TrackName"/>
   </element>
-  <element name="Language" path="\Segment\Tracks\TrackEntry\Language" id="0x22B59C" type="string" default="eng" minOccurs="1" maxOccurs="1">
+  <element name="Language" path="\Segment\Tracks\TrackEntry\Language" id="0x22B59C" type="string" length="not 0" default="eng" minOccurs="1" maxOccurs="1">
     <documentation lang="en" purpose="definition">Specifies the language of the track in the Matroska languages form;
 see (#language-codes) on language codes.
 This Element **MUST** be ignored if the LanguageIETF Element is used in the same TrackEntry.</documentation>
     <extension type="libmatroska" cppname="TrackLanguage"/>
   </element>
-  <element name="LanguageIETF" path="\Segment\Tracks\TrackEntry\LanguageIETF" id="0x22B59D" type="string" minver="4" maxOccurs="1">
+  <element name="LanguageIETF" path="\Segment\Tracks\TrackEntry\LanguageIETF" id="0x22B59D" type="string" length="not 0" minver="4" maxOccurs="1">
```

Comment:
We could add a restriction on the length which would be 3 for Matroska 1-3 and `>= 3 & <=6` for Matroska 4

Reply:
That doesn't make sense.

1. LanguageIETF wasn't introduced until v4.
2. The minimum length for a valid IETF BCP 47 language tag is two characters as all ISO 639 alpha 2 codes are valid (e.g. a simple `en`).
3. There is no maximum length for valid IETF BCP 47 language tags. Due to their "extensions" and "private use" sub-parts, they can have arbitrary length.

For the sake of not confusing anyone, just leave out any length restriction. Just state that it must be a valid IETF BCP 47 language tag.

Reply:
Indeed, only the v4 constraint makes sense.

Having machine readable constraints make it easier to validate files on reading/writing (and in mkvalidator) without knowing too much of the underlying data (which would be a deeper analysis than the container level).

As you said, it can be very lengthy so only `not 0` is possible here.